### PR TITLE
Add custom_packages role

### PR DESCRIPTION
**Pull request by santos-lucas**
## Describe your changes

Creation of custom_packages role, to manage the installation of packages that are not already handled by other roles.

## Checklist before requesting a review
- [x] Document introduced changes / variables in role README.md if any
- [x] Increment role version in vars/main.yml (a.b.c: +1 to b for a feature added, +1 to c for a bug fix)
- [x] Update changelog inside role README.md
- [x] If not present, please also add your name in the related collection authors list in galaxy.yml

### CODE REVIEW

File: collections/infrastructure/roles/custom_packages/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+- name: "Install sequana packages"
+  yum:
+    name: "{{sequana_packages[item]}}"
```
Comment (hmeScaler) [inline collections/infrastructure/roles/custom_packages/tasks/main.yml:3]: maybe 
```suggestion
    name: "{{custom_packages[item]}}"
```

**Comment by santos-lucas** [comment]
You are right @hmeScaler , many thanks to pointing that out.

I guess I got the fisrt version that I did for some sequana cluster, just fixed that + the other mentions with "sequana"

File: collections/infrastructure/roles/custom_packages/README.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
+# Custom Packages
+
+## Description
+
+This Role simply allows to install custom packages on the system that 
+are not provideded to other bluebanquise roles.
```
Comment (oxedions) [inline collections/infrastructure/roles/custom_packages/README.md:6]: provideded -> typo

File: collections/infrastructure/roles/custom_packages/README.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+That can be usefull to keep all the customizations relating OS packages
+synced with BlueBanquise so you dont need to have extra steps in 
+configuring nodes.
+
+## Instructions
+
+This role is very simple and works based on ansible groups.
+I recoomend you to have on single file on the "all" group ( $HOME/bluebanquise/inventory/group_vars/all/addons/custom_packages.yml)
```
Comment (oxedions) [inline collections/infrastructure/roles/custom_packages/README.md:14]: recoomend -> typo

**Review by oxedions — COMMENTED** [review]
Dear @santos-lucas 

Many thanks for this new role ! :blush: 
It is very close to https://github.com/bluebanquise/community/tree/main/roles/generic_psf

I was wondering: why filtering  by groups inside the role ? I mean by that: isn't it better to let users set variable per group in the inventory, and just load the hostvars to get node packages to install ?

**Comment by santos-lucas** [comment]
Hello @oxedions ,

I currently do it like this, because I use this in a couple cluster with hetereogeneous compute nodes, so it makes easier to look into the custom_packages configuration for the whole cluster, otherwise I would need to navigate between the group_vars directories and makes even harder to compare something if needed.

It is kinda of the same thing as we have for the host_vars in bluebanquise, we use everything on a single file instead of using the host_vars directory. For this single pruporse of the "custom_packages" it makes life a little easier, even more for sysadmins that work on systems with BlueBanquise but are not too experienced with it.

I see that it seems a lttle weird with the BB structure, but that is why I created it like this.

**Comment by oxedions** [comment]
Dear @santos-lucas 

Thank you for this honest feedback.
This is fair, ok for me! :blush:

**Review by oxedions — APPROVED** [review]

**Pull request merged**